braindecode/samplers/base.py

```diff
@@ -0,0 +1,69 @@
+"""
+Sampler classes.
+"""
+
+# Authors: Hubert Banville <hubert.jbanville@gmail.com>
+#
+# License: BSD (3-clause)
+
+from torch.utils.data.sampler import Sampler
+from sklearn.utils import check_random_state
+
+
+class RecordingSampler(Sampler):
+    """Base sampler simplifying sampling from recordings.
+
+    Parameters
+    ----------
+    metadata : pd.DataFrame
+        DataFrame describing at least one of {subject, session, run}
+        information for each window in the BaseConcatDataset to sample examples
+        from. Normally obtained as the `metadata` propery of BaseConcatDataset.
```

Comment:
typo on "propery".
Also this is still not so clear to me, maybe you an also put an minimal example in the docstring? Does not have be something that runs by itself, like just show relevant lines for understanding what is in metadata and what would happen.